### PR TITLE
Better config specs

### PR DESCRIPTION
**Pull request by SpComb**
Move all config specs into one place, and fix the specs to cover the complete hash => schema => struct codepath, also taking into account the `DEFAULT_DATA`.

### CODE REVIEW

**Comment by SpComb** [comment]
The CLI error handling should still behave exactly the same with the new `Pharos::ConfigError`:

```
$ docker-compose run --rm -e DEBUG=true pharos-cluster up || echo $?
==> KONTENA PHAROS v1.0.1 (Kubernetes v1.10.2)
==> Reading instructions ...
==> Invalid configuration:
---
:network:
  :provider:
  - 'must be one of: weave calico'
11
```

**Comment by SpComb** [comment]
Rubocop is complaining about existing code:

```
SplatKeywordArguments: Do not use splat keyword arguments as a single Hash.
        @client.delete_resource(@resource, **options)
^^^^^^^^^ 
```

File: lib/pharos/config.rb [hunk 1]
```diff
@@ -18,6 +18,21 @@ class Config < Dry::Struct
 
     constructor_type :schema
 
+    # @param data [Hash]
```
Comment (jnummelin) [inline lib/pharos/config.rb]: <nitpicking>
param name should be `raw_data` :)
</nitpicking>

**Comment by jnummelin** [comment]
maybe merge/rebase with  master to make rubocop happy for this PR

**Pull request merged**